src/pybasic/settings.py

```diff
@@ -0,0 +1,52 @@
+"""Settings dataclass for BaSiC."""
+
+from dataclasses import dataclass
+
+# NOTE convert to enum?
+ESTIMATION_MODES = ["l0"]
+
+
+@dataclass
+class Settings:
```

Comment:
I think this looks great. Maybe we should consider using `pydantic` because that will give us some nice utility functions for traceability as well as type checking.

What I mean for traceability is that it has convenience functions for standard representations of data objects. So we can dump data to json or yaml or whatever, and then if we want to re-use the same settings we can just pass the file into the `pydantic` class to re-use those settings. This is something we are concerned with in production environments to reproduce results, but I could foresee it being useful when we start doing things like Napari as well.